### DISCUSSION TITLE
How to iterate over fieldset?

### DISCUSSION THREAD

**Discussion by elreeda** [edmundhung/conform#520] [Q&A]
I have a dynamic object and not sure always sure ill get the same keys. you can certainly iterate over a `fieldList` as shown in the [doc](https://conform.guide/complex-structures#array). but with `fieldSet` given the same example from the doc and I don't always get `zipcode`, `country` for example, how to iterate dynamically?

```js
function Example() {
  const [form, fields] = useForm();
  const address = fields.address.getFieldset();

  return (
    <form id={form.id}>
      {/* Set the name to `address.street`, `address.zipcode` etc. */}
      <input name={address.street.name} />
      <div>{address.street.errors}</div>
      <input name={address.zipcode.name} />
      <div>{address.zipcode.errors}</div>
      <input name={address.city.name} />
      <div>{address.city.errors}</div>
      <input name={address.country.name} />
      <div>{address.country.errors}</div>
    </form>
  );
}
```

**Comment by elreeda** [comment]
I have two solutions, first convert to an array and use `fieldList` or since I'm using zod and I can just extract the shape from the schema something like:

```js
const conditionsList = conditions.getFieldset()
{Object.keys(schema.shape).map((type) => {
  return (
    <Box
      key={type}
    >
      <Select
        label={type}
        defaultValue={conditionsList[type].initialValue}
        name={conditionsList[type].name}
        options={[
          { value: 'notRelevant', label: 'notRelevant' },
          { value: 'optional', label: 'optional' },
          { value: 'required', label: 'required' },
        ]}
      />
    </Box>
  );
})}
```
I went up with the second one but i'd love if there is another way better that these solutions.

**Reply by edmundhung** [reply]
I don't have a better solution unfortunately.

There is no reliable way to know the shape of the schema afaik. `schema.shape` will works for simple schema, but it will not work if the schema is more complex, e.g. using `.refine()`, `.transform()` or even `.or()`.
